### ISSUE TITLE
DataStore throws an exception parsing the value returned from type [AWSTimestamp]

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-flutter#814]
**Describe the bug**
DataStore throws an exception parsing the value returned from type `[AWSTimestamp]`. This is due to `TemporalTimestamp.fromSeconds` expecting an int, but receiving a double. This is only seen on Android.

example schema

```graphql
type MyModel @model {
  id: ID!
  timestampType: AWSTimestamp # <-- This works as expected
  timestampListType: [AWSTimestamp] # <-- This will cause an exception
}
```

**To Reproduce**
1. Create a model such as the one above in your schema
2. Attempt to save an object
```dart
var now = DateTime.now();
MyModel testModel = MyModel(
  timestampType: TemporalTimestamp(now),
  timestampListType: [TemporalTimestamp(now), TemporalTimestamp(now)],
);
await Amplify.DataStore.save(testModel);
```
3. Attempt to query that model
```dart
var models = await Amplify.DataStore.query(MyModel.classType);
```

**Expected behavior**
An exception should not be thrown

**Screenshots**
The following screen shot shows that AWSTimestamp is being returned as int, but a list of AWSTimestamp ([AWSTimestamp]) is being returned as double 

<img width="546" alt="Screen Shot 2021-08-18 at 10 07 59 AM" src="https://user-images.githubusercontent.com/20613561/129914609-fd773a2b-10c7-4029-9196-86626ea5afa8.png">

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[] iOS

**NOTE**
Prior to this issue being marked as resolved, the DataStore integration tests that are skipped due to this issue should be updated. 

See: https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_datastore/example/integration_test/model_type_test.dart

**Comment by Jordan-Nelson** [comment]
### Cause
The cause of this is that the values returned from the native side (on Android) are doubles. That can be seen in the screenshot below.

<img width="546" alt="Screen Shot 2021-08-18 at 10 07 59 AM" src="https://user-images.githubusercontent.com/20613561/129916877-c75d3c91-e4da-4979-9a7d-89bf8cfab373.png">

The code to parse the timestamp expects an int, which can be seen below.

https://github.com/aws-amplify/amplify-flutter/blob/924861a3dd0ce3a82df94a1646da188c2a5aeb06/packages/amplify_datastore_plugin_interface/lib/src/types/temporal/temporal_timestamp.dart#L35-L37

### Options to resolve issue:

#### 1. Update `TemporalTimestamp.fromSeconds` to accept a dynamic, and then convert to int if a double is received.
This is how List<int> is currently handled. This logic is in the code generated models. An example can be seen below.
```dart
  _intListType = (json['intListType'] as List<dynamic>)
      ?.map((dynamic e) => e is double ? e.toInt() : e as int)
      ?.toList(),
```

**Pros:** This is consistent with how List<int> is handled.
**Cons:** This is only an issue on Android so it seems like it should be handled on the Android side. Putting logic in dart to handle this feels a bit like a bandaid. My personal preference would be to have all platforms returns the same data over the method channel so that we can avoid platform specific logic in Dart. Testing code (unit testing) both in dart and kotlin/swift is easier when we have a strict contract that we are enforcing over the method channel.

#### 2. Add logic to convert this in the platform specific (kotlin) code in amplify-flutter
**Pros:** Platform specific logic stays in platform specific code.
**Cons:** ?

#### 3. Look into addressing this in Amplify-Android
I haven't confirmed this, but it seems like the data is not being serialized correctly for lists on Android. This might just be a bug with SerializedModel.

**Pros:** Would be a more robust fix. We could fix this issue for any other list type that needs it (List<int>) and potentially remove the logic in the codegen models to handle doubles for type List<int>. Handling that logic in dart feels like a bit like a band-aid.
**Cons: ?**

**Comment by b-cancel** [comment]
This is a bug report I just filed that includes a couple of bugs I found and the workarounds to each
after reading your bug report I think it could help
https://github.com/aws-amplify/amplify-flutter/issues/822

**Comment by HuiSF** [comment]
The fix for this issue has been released with [version 0.3.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v0.3.0).
